File: frontend/src/types/timeline.ts
/** Reference to a related Trade Action */
export interface TimelineActionRef {
  id: string;
  action_type: string;
  confirmed_at: string;
  price: string | null;
  quantity: string | null;
}

/** Reference to a related accepted Analysis */
export interface TimelineAnalysisRef {
  id: string;
  analysis_type: string;
  accepted_at: string | null;
  schema_name: string;
  schema_version: string;
}

/** One timeline event */
export interface TimelineEvent {
  id: string;
  session_id: string;
  event_type: string;
  occurred_at: string;
  created_at: string;
  summary: string | null;
  price: string | null;
  quantity: string | null;
  related_action: TimelineActionRef | null;
  related_analysis: TimelineAnalysisRef | null;
}

/** GET /api/trade-sessions/{id}/timeline */
export interface ListTimelineResponse {
  events: TimelineEvent[];
  total: number;
}

File: backend/app/api/schemas/timeline.py
"""Pydantic schemas for Timeline API (TP-1006)."""

from __future__ import annotations

from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel


class TimelineActionReference(BaseModel):
    id: str
    action_type: str
    confirmed_at: datetime
    price: Decimal | None
    quantity: Decimal | None


class TimelineAnalysisReference(BaseModel):
    id: str
    analysis_type: str
    accepted_at: datetime | None
    schema_name: str
    schema_version: str


class TimelineEventResponse(BaseModel):
    id: str
    session_id: str
    event_type: str
    occurred_at: datetime
    created_at: datetime
    summary: str | None
    price: Decimal | None
    quantity: Decimal | None
    related_action: TimelineActionReference | None
    related_analysis: TimelineAnalysisReference | None


class TimelineListResponse(BaseModel):
    events: list[TimelineEventResponse]
    total: int
